src/rust-2021/default-cargo-resolver.md

```diff
@@ -16,5 +16,156 @@ The new feature resolver no longer merges all requested features for
+This has historically worked because Cargo unified the features of `bstr` between the two packages.
+However, after updating to Rust 2021, the new resolver will build `bstr` twice, once with the default features (as a build dependency), and once with no features (as our normal dependency).
+Since `bstr` is now being built without the "unicode" feature, the `words_with_breaks` method doesn't exist, and the build will fail with an error that the method is missing.
+
+The solution here is to ensure that the dependency is declared with the features you are actually using.
+For example:
+
+```toml
+[dependencies]
+bstr = { version = "0.2.16", default-features = false, features = ["unicode"] }
+```
+
+In some cases, this may be a problem with a third-party dependency that you don't have direct control over.
+You can consider submitting a patch to that project to try to declare the correct set of features for the problematic dependency.
+Alternatively, you can add features to any dependency from within your own `Cargo.toml` file.
+For example, if the `bstr` example given above was declared in some third-party dependency, you can just copy the correct dependency declaration into your own project.
+The features will be unified, as long as they match the unification rules of the new resolver. Those are:
+
+* Features enabled on platform-specific dependencies for targets not currently being built are ignored.
+* Build-dependencies and proc-macros do not share features with normal dependencies.
+* Dev-dependencies do not activate features unless building a target that needs them (like tests or examples).
+
+A real-world example is using [`diesel`](https://crates.io/crates/diesel) and [`diesel_migrations`](https://crates.io/crates/diesel_migrations).
+These packages provide database support, and the database is selected using a feature, like this:
+
+```toml
+[dependencies]
+diesel = { version = "1.4.7", features = ["postgres"] }
+diesel_migrations = "1.4.0"
+```
+
+The problem is that `diesel_migrations` has an internal proc-macro which itself depends on `diesel`, and the proc-macro assumes its own copy of `diesel` has the same features enabled as the rest of the dependency graph.
+After updating to the new resolver, it fails to build because now there are two copies of `diesel`, and the one built for the proc-macro is missing the "postgres" feature.
+
+A solution here is to add `diesel` as a build-dependency with the required features, for example:
+
+```toml
+[build-dependencies]
+diesel = { version = "1.4.7", features = ["postgres"] }
+```
+
+This causes Cargo to add "postgres" as a feature for host dependencies (proc-macros and build-dependencies).
+Now, the `diesel_migrations` proc-macro will get the "postgres" feature enabled, and it will build correctly.
+
+The 2.0 release of `diesel` (currently in development) does not have this problem as it has been restructured to not have this dependency requirement.
+
+### Exploring features
+
```

Comment:
It would be good to add a once sentence introduction to this section explaining that this is useful in the case you need to debug an issue (which we only expect to happen rarely). Otherwise this section feels a bit bolted on to the rest of the guide.